core/src/main/java/org/infinispan/commands/topology/AbstractTopologyCommand.java

```diff
@@ -0,0 +1,43 @@
+package org.infinispan.commands.topology;
+
+import org.infinispan.commands.GlobalRpcCommand;
+import org.infinispan.factories.scopes.Scope;
+import org.infinispan.factories.scopes.Scopes;
+import org.infinispan.remoting.transport.Address;
+
+@Scope(Scopes.NONE)
+public abstract class AbstractTopologyCommand implements GlobalRpcCommand {
+
+   private final byte commandId;
+
+   protected transient Address origin;
+
+   AbstractTopologyCommand(byte commandId) {
+      this(commandId, null);
+   }
+
+   AbstractTopologyCommand(byte commandId, Address origin) {
```

Comment:
this constructor isn't required. `origin` will be set on the receiver site.

Reply:
It is set on the receiver side, however the address is also used for local invocations so we need this in place. I tried removing the `origin`, however it results in the testsuite hanging.

Reply:
ah! I see... the `TopologyManagementHelper` invokes the command directly. :+1: